### PR TITLE
[BugFix] close cursor after used

### LINKED ISSUE #9284: Leader cannot delete previous db when making checkpoint just after transfered
After a follower transfer to leader, it will roll the journal right away. Later Checkpointer would make a checkpoint and remove all the databases. We'll get a error log here

```
2022-07-27 16:35:56,361 ERROR (leaderCheckpointer|131) [Daemon.run():117] daemon thread got exception. name: leaderCheckpointer
com.sleepycat.je.LockTimeoutException: (JE 7.3.8) Lock expired. Locker 1853030856 125_leaderCheckpointer_Txn: waited for lock on database=_jeNameMap LockAddr:122734431 LSN=0x0/0x10918 type=WRITE grant=WAIT_NEW timeoutMillis=1000 startTime=1658910955360 endTime=1658910956361
Owners: [<LockInfo locker="315960109 -1_replayer_HandleLocker" type="READ"/>]
Waiters: [<LockInfo locker="1853030856 125_leaderCheckpointer_Txn" type="WRITE"/>]

        at com.sleepycat.je.txn.LockManager.makeTimeoutException(LockManager.java:1117) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.txn.LockManager.waitForLock(LockManager.java:606) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.txn.LockManager.lock(LockManager.java:345) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.txn.Txn.lockInternal(Txn.java:539) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.txn.Locker.lock(Locker.java:499) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.CursorImpl.lockLN(CursorImpl.java:3585) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.CursorImpl.lockLN(CursorImpl.java:3316) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.CursorImpl.lockLNAndCheckDefunct(CursorImpl.java:2138) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.CursorImpl.searchExact(CursorImpl.java:1950) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.CursorImpl.searchExact(CursorImpl.java:1896) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.DbTree.getDb(DbTree.java:1511) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.DbTree.lockNameLN(DbTree.java:909) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.DbTree.doRemoveDb(DbTree.java:1172) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.dbi.DbTree.dbRemove(DbTree.java:1224) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.Environment$1.runWork(Environment.java:996) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.Environment$DbNameOperation.runOnce(Environment.java:1216) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.Environment$DbNameOperation.run(Environment.java:1199) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.sleepycat.je.Environment.removeDatabase(Environment.java:1004) ~[starrocks-bdb-je-7.3.8.jar:?]
        at com.starrocks.journal.bdbje.BDBEnvironment.removeDatabase(BDBEnvironment.java:481) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJEJournal.deleteJournals(BDBJEJournal.java:239) ~[starrocks-fe.jar:?]
        at com.starrocks.leader.Checkpoint.runAfterCatalogReady(Checkpoint.java:176) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:60) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9284 

## Problem Summary(Required) ：

After a follower transfers to leader, it must close the database it was using when replayer quit.
Otherwise, as a new leader, by the time it make a snapshot, it will try to remove all the previous databases and fail.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|5|20.00%|[1637, 1638, 1639, 1641]|

**Review by chaoyli — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 1]
```diff
@@ -1647,6 +1656,7 @@ public boolean replayJournal(long toJournalId) throws JournalException {
         try {
             cursor = journal.read(replayedJournalId.get() + 1, toJournalId);
             hasLog = replayJournalInner(cursor, false);
+            cursor.close();
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1659]: this line is not needed, you will close it in finally block

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1659]: You're right. My mistake.

Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1659]:
```suggestion
```

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|4|00.00%|[1637, 1638, 1639, 1641]|

**Pull request merged**